torch/lib/c10d/reducer.cpp

```diff
@@ -1466,18 +1475,61 @@ void Reducer::ensure_prior_reduction_finished() {
   }
 }
 
+void Reducer::set_env_variables() {
+  // Environment variables
+  ddp_logging_data_->master_port = parse_env("MASTER_PORT");
+  ddp_logging_data_->master_addr = parse_env("MASTER_ADDR");
+  ddp_logging_data_->cuda_visible_devices = parse_env("CUDA_VISIBLE_DEVICES");
+  ddp_logging_data_->gloo_socket_ifname = parse_env("GLOO_SOCKET_IFNAME");
+  ddp_logging_data_->gloo_device_transport = parse_env("GLOO_DEVICE_TRANSPORT");
+  ddp_logging_data_->nccl_socket_ifname = parse_env("NCCL_SOCKET_IFNAME");
+  ddp_logging_data_->nccl_blocking_wait = parse_env("NCCL_BLOCKING_WAIT");
+  ddp_logging_data_->nccl_debug = parse_env("NCCL_DEBUG");
+  ddp_logging_data_->nccl_nthreads = parse_env("NCCL_NTHREADS");
+  ddp_logging_data_->nccl_ib_timeout = parse_env("NCCL_IB_TIMEOUT");
+}
+
+void Reducer::set_parameter_stats() {
+  ddp_logging_data_->num_parameters = 0;
+  ddp_logging_data_->parameter_size = 0;
+  for (const auto& t : replicas_[0]) {
+    ddp_logging_data_->num_parameters++;
+    ddp_logging_data_->parameter_size += t.numel() * t.element_size();
+  }
+}
+
+void Reducer::set_bucket_stats() {
+  for (const auto& bucket : buckets_) {
+    const auto& variables = bucket.replicas[0].variables;
+    int bucket_size = 0;
+    for (const auto& v : variables) {
+      bucket_size += v.numel() * v.element_size();
+    }
+    ddp_logging_data_->bucket_sizes.push_back(bucket_size);
+  }
+}
+
 void Reducer::set_construction_logging_data(
   const std::string& module_name,
   const std::vector<int>& device_ids,
   int output_device,
   bool broadcast_buffers
 ) {
+// Data that can be got during DistributedDataParallel construction time
   ddp_logging_data_->module_name = module_name;
+  ddp_logging_data_->world_size = process_group_->getSize();
+  ddp_logging_data_->rank = process_group_->getRank();
+  ddp_logging_data_->iteration = 0;
+  ddp_logging_data_->dtype = std::string(replicas_[0][0].dtype().name());
+
+  set_parameter_stats();
+  set_bucket_stats();
```

Comment:
Would we need to reset these after buckets are rebuilt?

Reply:
yes, rebuild bucket sizes is added in the next PR where it needs to be recalculated